addons/block_code/ui/block_canvas/serialized_block_tree_node.gd

```diff
@@ -1,10 +1,16 @@
 class_name SerializedBlockTreeNode
 extends Resource
 
-@export var serialized_block: SerializedBlock
+@export var name: StringName
+@export var position: Vector2
```

Comment:
Another thing here: Why does every `SerializedBlockTreeNode` have its position serialized? Only the top level block really needs its position saved. Anyways, it still works fine, with nested blocks having position (0,0), and I wouldn't change it here unless you really want to because it allows you to make very minimal changes to `BlockCanvas`.

Reply:
> Another thing here: Why does every `SerializedBlockTreeNode` have its position serialized? Only the top level block really needs its position saved. Anyways, it still works fine, with nested blocks having position (0,0), and I wouldn't change it here unless you really want to because it allows you to make very minimal changes to `BlockCanvas`.

Yes, I thought about this. For a snapped/attached block, as soon as the user detaches it by dragging it to the canvas, it will need a position. So I thought that it is better to keep the position as property to all the saved blocks, and just ignore it if they are attached. What do you think?

Reply:
Yep, that's totally fine. It makes the code a little simpler in this case!